### PR TITLE
rt: close ticket with proper owner

### PR DESCRIPTION
**Pull request by pazembrz**

### CODE REVIEW

File: inspirehep/utils/tickets.py [hunk 1]
```diff
@@ -214,34 +221,75 @@ def get_queues():
...
+
+    Args:
+        email(str): Email of user to look for.
+
+    Returns:
+        dict: dictionary of the user {"id":"user/<id>", "Name":"<name>", "EmailAddress": "<e-mail>"}.
+
+    """
+    def _parse_users_with_emails(lines):
+        data = zip(lines[0::3], lines[1::3], lines[2::3])
```
Comment (michamos) [inline inspirehep/utils/tickets.py:246]: nice API :grimacing:

File: inspirehep/utils/tickets.py [hunk 2]
```diff
@@ -214,34 +221,75 @@ def get_queues():
...
+
+    def _get_rt_users():
+        _lines = query_rt('user', fields="Name,EmailAddress")
+        _users_map = _parse_users_with_emails(_lines)
+        if not _users_map:
+            raise EmptyResponseFromRT
+        return _users_map
+
+    users_map = _get_from_cache("rt_users_with_emails", _get_rt_users, force_update)
+    result = filter(lambda user: user['EmailAddress'] == email, users_map)
```
Comment (michamos) [inline inspirehep/utils/tickets.py:261]:
```suggestion
    result = [user for user in users_map if user['EmailAddress'] == email]
```
is more readable

File: inspirehep/utils/tickets.py [hunk 3]
```diff
@@ -214,34 +221,75 @@ def get_queues():
...
+        data = zip(lines[0::3], lines[1::3], lines[2::3])
+        users = []
+        for user_data in data:
+            if user_data and all(user_data):
+                users.append(dict([map(unicode.strip, element.split(':', 1)) for element in user_data]))
+        return users
+
+    def _get_rt_users():
+        _lines = query_rt('user', fields="Name,EmailAddress")
+        _users_map = _parse_users_with_emails(_lines)
```
Comment (michamos) [inline inspirehep/utils/tickets.py:255]: misnomer, this is not a mapping, why not simply `users_data` (what's with the `_` BTW)?

File: inspirehep/utils/tickets.py [hunk 4]
```diff
@@ -214,34 +221,75 @@ def get_queues():
...
+    def _parse_users_with_emails(lines):
+        data = zip(lines[0::3], lines[1::3], lines[2::3])
+        users = []
+        for user_data in data:
+            if user_data and all(user_data):
+                users.append(dict([map(unicode.strip, element.split(':', 1)) for element in user_data]))
+        return users
+
+    def _get_rt_users():
+        _lines = query_rt('user', fields="Name,EmailAddress")
```
Comment (michamos) [inline inspirehep/utils/tickets.py:254]: just to make sure: does this list contain only priviledged users, or also all users who ever sent an email to RT? the latter is huge.

File: inspirehep/utils/tickets.py (hunk 1 above)
Comment (pazembrz) [reply inspirehep/utils/tickets.py:246]: You cannot do much more with this beautiful response from RT :P

File: inspirehep/utils/tickets.py (hunk 2 above)
Comment (pazembrz) [reply inspirehep/utils/tickets.py:261]: I prefer filter IMHO but ok :+1:

File: inspirehep/utils/tickets.py (hunk 3 above)
Comment (pazembrz) [reply inspirehep/utils/tickets.py:255]: :+1:  
I added `_`  as my IDE is complaining that those are shadowing vars from outer scope. It's not breaking anything but I do not like to add more "warnings" to this beautiful code on next ;)

File: inspirehep/utils/tickets.py (hunk 4 above)
Comment (pazembrz) [reply inspirehep/utils/tickets.py:254]: When I ran this query it only returned a dozen or so results. So as I assume it returns only privileged users, but it's difficult to say as /search/user looks to be undocumented :/ https://rt-wiki.bestpractical.com/wiki/REST

**Pull request closed without merging**